_inc/client/components/module-settings/index.jsx

```diff
@@ -341,17 +347,26 @@ export let TiledGallerySettings = React.createClass( {
 TiledGallerySettings = ModuleSettingsForm( TiledGallerySettings );
 
 export let PostByEmailSettings = React.createClass( {
+	regeneratePostByEmailAddress() {
+		this.props.regeneratePostByEmailAddress();
+	},
 	render() {
 		return (
 			<form onSubmit={ this.props.onSubmit } >
 				<FormFieldset>
-					<span> { '(Currently does not work)' } </span>
-					<ModuleOptionTextInput option_name={ 'post_by_email_address' }
-						{ ...this.props }
-						label={ __( 'Email Address' ) } />
-					<FormButton>{ __( 'Regenerate address' ) }</FormButton>
+					<FormFieldset>
```

Comment:
Uh oh, maybe take another shot at this? :) You seem to have `FormFieldset` here twice.